recipes/pytest-pyvista/meta.yaml

```
@@ -0,0 +1,48 @@
+{% set name = "pytest-pyvista" %}
+{% set version = "0.1.8" %}
+
+package:
+  name: {{ name|lower }}
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/p/pytest_pyvista/pytest_pyvista-{{ version }}.tar.gz
+  sha256: 43e6f0a85b6ff23aff51b7f750528de0fdda6da2011a94b636f05d8498fe38a0
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  number: 0
+
+requirements:
+  host:
+    - pip
+    - python >=3.8
+    - flit >=3.2,<4
+  run:
+    - python >=3.8
+    - pytest >=3.5.0
+
```

Do we need to have PyVista in the `run` section? It will be needed for import, and I'm not sure if this is a problem conda-forge recipe.  It isn't in the pip install dependencies because of the weirdness of vtk installation having multiple possible package names.

---

If `pytest-pyvista` imports pyvista, and it seems that it does https://github.com/pyvista/pytest-pyvista/blob/main/pytest_pyvista/pytest_pyvista.py#L8, then you need `pyvista` in the `run` section 👍

---

There shouldn't be a problem if `pyvista` is missing in the pip dependencies, as when`pip` is used to install the built `pytest-pyvista` package here it uses the `--no-deps` option, delegating the responsibility to `conda-build` to ensure that all the necessary runtime dependencies are pre-installed.

Conda will figure out the appropriate build of `vtk` to install given the install environment e.g., qt or egl et al

I don't think this is a problem ... are we willing to try and see?

---

Also, I pretty much exclusively use conda to install `pyvista` and next to never with `pip`.

I'm guessing you are the other way around?

---

I don't know about conda, but I will lay out the issue with pip and vtk.  pyvista imports vtk.  You can install `vtk` or `vtkosmesa`.  If you install `vtkosmesa` first, it might be a surprise when you install `pytest-pyvista` that it will install `pyvista` which will install `vtk`.  We had decided to remove `pyvista` from dependencies as a result.  Ideally python would instead provide default optional_dependencies, which would prevent this problem.  We could have `pip install pyvista[openmesa]` to get `vtk-osmesa` or just `pip install pyvista` to get regular vtk.  Similar for this package.  How does conda handle this?

---

@MatthewFlamm Thanks for explaining, this is really helpful insight, as I'm pretty new to `pyvista` and `vtk`.

Okay, so typically I leave the `conda` resolver to determine the `vtk` build bindings, but you can also be explicit as per your example with `pip` e.g., if you know that you want the `osmesa` build of `vtk` then you can do `conda install -c conda-forge pyvista "vtk=*=osmesa_*"`

The `vtk=*=osmesa_*` spec is of the form `<package>=<version>=<build>`, so here we're requesting for the latest version of `vtk` with the latest `osmesa` build.

Again, if you wanted the QT build you'd similarly do `vtk=*=qt_*`.

Does this help?

---

I see now, whoever packaged vtk for conda-forge thankfully did not follow the model vtk uses for pip.  So it seems like the problem doesn't exist here.  Should be no problem to include then.

---

Yeah, I *think* so ...

Personally, I say we give it a whirl and see how it behaves.

What do you guys think?

---

> I see now, whoever packaged vtk for conda-forge thankfully did not follow the model vtk uses for pip.

BTW, folks usually compare conda with pip but it is important to remind that conda is a language agnostic package manager. Think apt, yum, choco, brew, etc, not pip.

---

Thank you for your comments. First, let's add PyVista and see what happens.
```suggestion
    - pyvista

```